File: ch6/akkamaid-java/src/main/java/com/akkademy/ArticleParseActor.java
package com.akkademy;

import akka.actor.AbstractActor;
import akka.actor.Status;
import akka.japi.pf.ReceiveBuilder;

public class ArticleParseActor extends AbstractActor {
    private ArticleParseActor() {
        receive(ReceiveBuilder.
                match(String.class, html -> {
                            System.out.println(ArticleParser.apply(html).orElse("empty?"));

                            ArticleParser.apply(html).
                                    onSuccess(body -> sender().tell(body, self())).
                                    onFailure(t -> sender().tell(new Status.Failure(t), self()));
                        }
                ).
                matchAny(x -> {
                    System.out.println("GOT A MSG!!! " + x);
                }).
                build());
    }
}

File: ch6/akkademaid-scala/src/main/scala/com/akkademy/ArticleParseActor.scala
package com.akkademy

import akka.actor.Actor
import com.akkademy.ParseArticle

class ArticleParseActor extends Actor{
  override def receive: Receive = {
    case htmlString: String =>
      val body: String = ArticleParser(htmlString)
      sender() ! body
    case _ =>
      println("msg!")
  }
}

object ArticleParser {
  def apply(html: String) : String =
    de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html)
}